controllers/testutils_test.go

```diff
@@ -29,31 +31,82 @@ const (
 )
 
 var (
-	defaultDuration = metav1.Duration{Duration: time.Minute * 5}
+	defaultDuration = metav1.Duration{Duration: time.Minute * 5}     // treat as const
+	letterRunes     = []rune("abcdefghijklmnopqrstuvwxyz1234567890") // treat as const
 )
 
-func defaultAppGroup(targetNamespace string) *v1alpha1.ApplicationGroup {
+func objKeyBuilder(name, namespace string) types.NamespacedName {
```

Comment:
I think this function can be removed as someone can just right `types.NamespacedName{Name:...., Namespace:...}` and it accomplishes the same thing. Adding a function for this is redundant

Reply:
Fixed